packages/notification-service/src/invites/invites.ts

```diff
@@ -7,27 +8,36 @@ import { getContractKit } from '../util/utils'
 
 const TAG = 'INVITES'
 
-export async function handleInvites() {
-  const kit = await getContractKit()
-  const fromBlock = getLastInviteBlockNotified() + 1
-  if (fromBlock <= 0) {
+function updateLastInviteBlock(fromBlock: number, events: EventLog[]) {
+  if (events.length === 0) {
     return
   }
-  console.debug(TAG, `Starting to fetch invites from block ${fromBlock}`)
+  const maxBlock = events.reduce((max, event) => Math.max(max, event.blockNumber), fromBlock)
+  setLastInviteBlockNotified(maxBlock)
```

Comment:
Should we be keeping track of `lastInviteBlockNotified` or rather `lastBlockChecked`? It looks like we are only updating the back end of the range whenever we have events that we've detected. However, we are polling every 10 seconds. From what I can see, that means we are keeping the range unnecessarily large by not acknowledging that we've already checked some blocks and determined that there were no relevant invites in them.

Am I misunderstanding something?

Reply:
Well, the range is only large as long as there are no events returned. When there are no events the request returns very fast (events are indexed by the fromBlock), so I don't see a big issue with that.

Since we're making requests without specifying a `to` block it defaults to `latest`, so we don't actually know which is the last block checked. We could ask for the latest block beforehand and set it explicitly, but that would require making an extra query. Not a big deal, but also not sure if it's necessary.

If you feel strongly about this I can change it though, I don't think it makes a big difference either way.

Reply:
I don't know enough about the `escrow.getPastEvents` query to have a strong opinion.

Let's take an extreme example of someone who's used Valora for a year but and never invited anyone. In your opinion, there's no material performance difference in querying from -1 to latest every time the polling function runs vs. updating that -1 to whatever was checked last? I guess if events are indexed based off of user address I can see how that's the case but otherwise I'm not sure.

Reply:
We're using `getPastEvents` for all withdrawal events, not per-user. So the max number of events returned by the query is the number of events that fit in two blocks (since we poll every 10 seconds) unless Forno gets backed up I guess. This is regardless of how long each individual user has been using the network.

There is little performance difference in querying from `-1` to `latest` or from something else to `latest` if there are no events in that range, but since we're storing the latest notified block we're always going to be querying using the latest notified block as the `fromBlock`, so that limits the number of events that might be returned.

Reply:
Ah I see - for some reason I was thinking this was running on Valora and would be specific to each user. I realize now that `lastInviteBlockNotified` refers to the last time any escrow withdrawal happened across the network so in theory this range will stay tighter the more frequently withdrawals are happening.